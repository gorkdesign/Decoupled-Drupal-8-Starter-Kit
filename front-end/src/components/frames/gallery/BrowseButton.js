import React from 'react';
import PropTypes from 'prop-types';

export const messages = {
  none: 'No files chosen',
  one: '1 file chosen',
  many: 'files chosen',
};

export const calculateMessage = (totalFiles, totalBytes) => {
  let message = messages.none;
  if (totalFiles >= 1) {
    message = totalFiles === 1 ? messages.one : message = `${totalFiles} ${messages.many}`;
  }

  message += (totalBytes < 1000000) ?
    ` (${Math.floor(totalBytes / 1000)}KB)` :
    ` (${Math.floor(totalBytes / 1000000)}MB)`;
  return message;
};

const BrowseButton = ({ totalFiles = 0, totalBytes = 0, render }) => (
  <div className="browseContainer">
    {render()} {calculateMessage(totalFiles, totalBytes)}
  </div>
);

BrowseButton.propTypes = {
  totalFiles: PropTypes.number,
  totalBytes: PropTypes.number,
  render: PropTypes.func.isRequired,
};

BrowseButton.defaultProps = {
  totalBytes: 0,
  totalFiles: 0,
};

export default BrowseButton;
