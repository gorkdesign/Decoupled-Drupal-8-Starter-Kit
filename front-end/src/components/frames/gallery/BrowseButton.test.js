import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import BrowseButton, { calculateMessage, messages } from './BrowseButton';

// snapshot tests.
const uploadInitiated = false;
const initialPropsState = {
  totalFiles: 0,
  totalBytes: 0,
  render: () => (
    <button
      type="button"
      onClick={jest.fn()}
      disabled={uploadInitiated}
    >
  Choose Files
    </button>),
};

const generateShallow = (props = initialPropsState) =>
  shallow(<BrowseButton {...props} />);


describe('BrowseButton', () => {
  it('should render correctly', () => {
    const actual = shallowToJson(generateShallow());
    expect(actual).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const props = {
      ...initialPropsState,
      render: jest.fn(),
    };
    const actual = shallowToJson(generateShallow(props));
    expect(actual).toMatchSnapshot();
  });

  it('should generate No files message', () => {
    const actual = calculateMessage(0, 0);
    const expected = `${messages.none} (0KB)`;
    expect(actual).toEqual(expected);
  });

  it('should generate 1 file message', () => {
    const actual = calculateMessage(1, 0);
    const expected = `${messages.one} (0KB)`;
    expect(actual).toEqual(expected);
  });

  it('should generate 5 file messages w/ 10kb', () => {
    const actual = calculateMessage(5, 10000);
    const expected = `5 ${messages.many} (10KB)`;
    expect(actual).toEqual(expected);
  });

  it('should generate 5 file messages w/ 10MB', () => {
    const actual = calculateMessage(5, 10000000);
    const expected = `5 ${messages.many} (10MB)`;
    expect(actual).toEqual(expected);
  });
});

