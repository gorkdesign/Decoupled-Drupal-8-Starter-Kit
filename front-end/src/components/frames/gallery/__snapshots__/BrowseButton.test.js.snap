// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseButton should render correctly 1`] = `
<div
  className="browseContainer"
>
  <button
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Choose Files
  </button>
   
  No files chosen (0KB)
</div>
`;

exports[`BrowseButton should render correctly 2`] = `
<div
  className="browseContainer"
>
   
  No files chosen (0KB)
</div>
`;
